src/plugins/intel_gpu/src/kernel_selector/tensor_type.cpp

```diff
@@ -550,6 +550,27 @@ DataTensor DataTensor::FlattenEverything() const {
     return res;
 }
 
+DataTensor DataTensor::SwapXYSpatials(bool swapXY = false) const {
+    DataLayout l = Tensor::bfyx;
+
+    auto x = X();
+    auto y = Y();
+
+    if (!swapXY || layout != l || x.pad.Total() != 0) {
+        return *this;
+    }
```

Comment:
I think this is dangerous code. It is called multiple times and some may be swapped and some may not be swapped.

Reply:
Actually, it will be called when swapXY is true. In other cases, it is same with calling convert_data_tensor() function.